packages/flutter/lib/src/material/refresh_indicator.dart

```diff
@@ -594,3 +586,170 @@ class RefreshIndicatorState extends State<RefreshIndicator> with TickerProviderS
+///    with [BouncingScrollPhysics] regardless of platform.
+///
+/// In a typical application, this sliver should be inserted between the app bar
+/// sliver such as [SliverAppBar] and your main scrollable
+/// content's sliver like: [CustomScrollView].
+///
+/// {@tool dartpad}
+/// When the user scrolls past [refreshTriggerPullDistance],
+/// this sample shows material pull to refresh indicator for 1 second and
+/// adds a new item to the top of the list view.
+///
+/// ** See code in examples/api/lib/material/refresh_indicator/sliver_refresh_control.0.dart **
+/// {@end-tool}
+///
+/// See also:
+///
+///  * [CustomScrollView], a typical sliver holding scroll view this control
+///    should go into.
+///  * [RefreshIndicator], a Material Design version of the pull-to-refresh
+///    paradigm. This widget works differently than [SliverRefreshControl]
+///    because instead of being part of the scrollable and actively occupies
+///    scrollable space. it goose overlay on top of the scrollable.
+///
+class SliverRefreshIndicator extends BaseSliverRefreshControl {
+  /// Create a new refresh control for inserting into a list of slivers.
+  ///
+  /// The [refreshTriggerPullDistance] and [refreshIndicatorExtent] arguments
+  /// must not be null and must be >= 0.
+  ///
+  /// The [builder] argument may be null, in which case no indicator UI will be
+  /// shown but the [onRefresh] will still be invoked. By default, [builder]
+  /// shows a [CupertinoActivityIndicator].
+  ///
+  /// The [onRefresh] argument will be called when pulled far enough to trigger
+  /// a refresh.
+  const SliverRefreshIndicator({
+    super.key,
+    super.refreshTriggerPullDistance,
+    super.refreshIndicatorExtent,
+    super.builder,
+    super.onRefresh,
+    this.backgroundColor,
+    this.color,
+    this.semanticsLabel,
+    this.semanticsValue,
+    this.radius = _progressIndicatorRadius,
+    this.strokeWidth = _defaultStrokeWidth,
+  });
+
+  /// The progress indicator's background color.
+  final Color? backgroundColor;
+
+  ///{@macro flutter.progress_indicator.ProgressIndicator.color}
+  final Color? color;
+
+  ///{@macro flutter.progress_indicator.ProgressIndicator.semanticsLabel}
+  final String? semanticsLabel;
+
+  ///{@macro flutter.progress_indicator.ProgressIndicator.semanticsValue}
+  final String? semanticsValue;
+
+  ///{@macro flutter.material.CircularProgressIndicator.strokeWidth}
+  ///
+  /// Defaults to 2.5px. Must be positive and cannot be null.
+  final double strokeWidth;
+
+  /// Determinate the radius of the [RefreshProgressIndicator].
+  ///
+  /// Defaults to 50px. Must be positive and cannot be null.
+  final double radius;
+
+  static const double _defaultStrokeWidth = 2.5;
+  static const double _progressIndicatorRadius = 25.0;
+
+  Widget _indicator([double? progress]) {
+    return SizedBox(
+      width: radius * 2,
+      height: radius * 2,
+      child: FittedBox(
```

Comment:
The material refresh indicator does not resize when you overscroll I think? If you use a `FittedBox` here it will expand when the user overscrolls and shrink when the scroll view bounces back.

Reply:
This fit the circular indicator in the refresh indicator, other way the look will be like:

<img src="https://user-images.githubusercontent.com/59793332/173525841-332d0c0d-0f45-4de1-99c9-e8d4c477d318.jpg" width="180px"/>